src/adapter/src/coord/sequencer.rs

```diff
@@ -2133,23 +2134,17 @@ impl<S: Append + 'static> Coordinator<S> {
         }
 
         let source_ids = source.depends_on();
-        let timeline = self.validate_timeline(source_ids.clone())?;
-        // Queries are independent of the logical timestamp iff there are no referenced
-        // sources or indexes and there is no reference to `mz_now()`.
-        let timestamp_independent = source_ids.is_empty() && !source.contains_temporal();
+        let mut timeline_context = self.validate_timeline_context(source_ids.clone())?;
+        if matches!(timeline_context, TimelineContext::TimestampIndependent)
+            && source.contains_temporal()
+        {
+            // If the source IDs are timeline independent but the query contains temporal functions,
+            // then the timeline context needs to be upgraded to timeline dependent.
+            timeline_context = TimelineContext::TimestampDependent;
+        }
         let in_immediate_multi_stmt_txn = session.transaction().is_in_multi_statement_transaction()
             && when == QueryWhen::Immediately;
 
-        if in_immediate_multi_stmt_txn {
```

Comment:
Is this feature now a side effect of some of the new stuff?

Reply:
Instead of clearing out the existing ops and inserting new one, `session.add_transaction_ops` knows how to explicitly handle all scenarios where there is existing ops.